File: top_interview_questions/1_easy/2_strings/valid_anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        counter_s = self.countChars(s)
        counter_t = self.countChars(t)

        return counter_s == counter_t

    def countChars(self, s):
        counter = {}

        for char in s:
            if char in counter:
                counter[char] += 1
            else:
                counter[char] = 1

        return counter

File: top_interview_questions/1_easy/2_strings/valid_anagram/solution.go
package main

import "fmt"

func main() {
	fmt.Println(isAnagram("nagaram", "anagram"))
	fmt.Println(isAnagram("rat", "car"))
}

func isAnagram(s, t string) bool {
	if len(s) != len(t) {
		return false
	}

	c1 := counter(s)
	c2 := counter(t)

	if len(c1) != len(c2) {
		return false
	}

	for k := range c1 {
		v1, v2 := c1[k], c2[k]
		if v1 != v2 {
			return false
		}
	}

	return true
}

func counter(s string) map[rune]int {
	c := make(map[rune]int)

	for _, char := range s {
		c[char]++
	}

	return c
}
